Documentation/dash-hld.md

```diff
@@ -0,0 +1,180 @@
+
+A contemporary DASH "SmartNIC" may consist of many complex hardware components including multi-core System On A Chip (SoC) ASICs, and the associated software. For simplicity, the software for such systems which interfaces to the SAI layer is collectively called the "ASIC driver." More importantly, the vendor SAI library will hide all details and present a uniform interface.
+
+### DASH capable ASICs
+These comprise the main dataplane engines and are the core of what are variously called SmartNICs, DPUs, IPUs, NPUS, etc. The actual cores may be ASICs, SoCs, FPGAs, or some other high-density, performant hardware.
+
+## Detailed architectures
+
+### DASH NOS single DPU on NIC
+
+![dash-single-dpu-architecture](images/hld/dash-single-dpu-architecture.svg)
+
+
+### DASH appliance architecture
+
+#### High level architecture
+
+![dash-high-level-appliance](images/hld/dash-high-level-appliance.svg)
+
+#### Low level architecture
+
+![dash-appliance-architecture](images/hld/dash-appliance-architecture.svg)
+
+
+### DASH smart switch architecture
+
+#### High level architecture
+
+![dash-high-level-smart-switch](images/hld/dash-high-level-smart-switch.svg)
+
+#### Low level architecture
+
+![dash-smart-switch-architecture](images/hld/dash-smart-switch-architecture.svg)
+
+
+## High-Level Design
+
+![dash-high-level-design](./images/hld/dash-high-level-design.svg)
+
+The system architecture for SONiC-DASH relies upon the [SONiC system architecture](https://github.com/Azure/SONiC/wiki/Architecture) and adds a *new docker container* in the user space named **dash container** to create the functional component for DASH.  
+
+In the **sync-d container** we will add **sai api DASH** (as opposed to *sai api* in the original SONiC architecture).  
+
+The *DPU/IPU/SmartNic* hardware will run a separate instance of SONiC-DASH on the hardware.  
+
+The component interactions will be executed as a new user space container implementation; relying on the existing SONiC infrastructure and components to interact as they normally would.  
+
+The functionality of the new *dash container* in the user space is to receive content from the Software Defined Networking (SDN) controller to control setup for the overlay configurations. DASH receives the objects, translates them with a **gNMI agent**, provides them to the *SONiC OrchAgent* for further translation onto the dataplane via the **SAI database**. 
+
+> [!NOTE] 
+> @lihuay @lguohan @prsunny - would you review and/or improve this write-up?
+
+## DASH deployment options
+
+The following figure is a simplified representation of DASH deployment in a datacenter. 
+
+![dash-simplified-physical-deployment-example](images/hld/dash-simplified-physical-deployment-example-v2.svg) 
+
+## Repositories
+
+TBD
+
+
+## References
+
+- [Glossary](https://github.com/Azure/DASH/wiki/Glossary)
+- [SAI](../SAI)
+- [Test](../test)
+- [SDN Features Packet Transforms](SDN-Features-Packet-Transforms.md)
```

Comment:
[SAI Thrift PR](https://github.com/opencomputeproject/SAI/pull/1325)
[P4](https://opennetworking.org/p4) and [P4 working group](https://p4.org/working-groups)
[PINS](opennetworking.org/pins)
[PNA Consortium Spec](https://p4.org/p4-spec/docs/PNA-v0.5.0.html)
[IPDK](https://ipdk.io/) and [IPDK GitHub](https://github.com/ipdk-io/ipdk-io.github.io)
[bmv2 - behavioral model v2](https://github.com/p4lang/behavioral-model)
[DPDK](https://www.dpdk.org)

Reply:
Done